### PR TITLE
CORE-6422 base api cleanup

### PR DESCRIPTION
**Pull request by blsemo**
Removing things that are unused from the api base package.
Moving things to a runtime base package where possible.

API PR: https://github.com/corda/corda-api/pull/548

### CODE REVIEW

**Review by lankydan — DISMISSED** [review]

File: components/http-rpc-gateway-comp/src/main/kotlin/net/corda/components/rpc/internal/HttpRpcGatewayEventHandler.kt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package net.corda.components.rpc.internal
 
+import net.corda.utilities.NetworkHostAndPort
```
Comment (vkolomeyko) [inline components/http-rpc-gateway-comp/src/main/kotlin/net/corda/components/rpc/internal/HttpRpcGatewayEventHandler.kt]: Minor: Would be good to have imports aligned in alpabetical order.

**Review by vkolomeyko — DISMISSED** [review]
LGTM

File: libs/http-rpc/http-rpc-server-impl/build.gradle [hunk 2]
```diff
@@ -10,7 +10,7 @@ dependencies {
     compileOnly 'org.osgi:osgi.core'
     implementation project(":osgi-framework-api")
 
-    implementation project(":libs:http-rpc:http-rpc")
+implementation project(":libs:http-rpc:http-rpc")
```
Comment (ronanbrowne) [inline libs/http-rpc/http-rpc-server-impl/build.gradle:13]: formatting

File: libs/base/build.gradle [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+
+dependencies {
+    compileOnly 'org.osgi:osgi.annotation'
+    compileOnly 'org.osgi:osgi.core'
+
+    implementation platform("net.corda:corda-api:$cordaApiVersion")
+    implementation 'net.corda:corda-base'
+    implementation 'net.corda.kotlin:kotlin-stdlib-jdk8-osgi'
+    // Concluded this is the one acceptable dependency in addition to kotlin.
```
Comment (ronanbrowne) [inline libs/base/build.gradle]: Was this intentionally left in or a note to self in development?

File: libs/http-rpc/http-rpc-server-impl/build.gradle (hunk 2 above)
Comment (ronanbrowne) [inline libs/http-rpc/http-rpc-server-impl/build.gradle]: formatting

**Review by ronanbrowne — CHANGES_REQUESTED** [review]

**Review by ronanbrowne — APPROVED** [review]
LGTM (build changes reviewed)

**Review by lankydan — APPROVED** [review]

**Pull request merged**